File: website/app/Models/KasModel.php
<?php

namespace App\Models;

use CodeIgniter\Model;

class KasModel extends Model
{
    protected $table            = 'kas';
    protected $primaryKey       = 'id';
    protected $useAutoIncrement = true;
    protected $returnType       = 'array';
    protected $useSoftDeletes   = false;
    protected $protectFields    = true;
    protected $allowedFields    = [
        'karang_taruna_id',
        'jenis', 
        'nominal', 
        'keterangan', 
        'tanggal', 
        'dibuat_oleh'
    ];

    protected $useTimestamps = true;
    protected $dateFormat    = 'datetime';
    protected $createdField  = 'created_at';
    protected $updatedField  = 'updated_at';

    // Mendapatkan total kas
    public function getTotalSaldo($karangTarunaId = null)
    {
        $db = \Config\Database::connect();
        
        $pemasukanBuilder = $db->table($this->table)->selectSum('nominal')->where('jenis', 'pemasukan');
        $pengeluaranBuilder = $db->table($this->table)->selectSum('nominal')->where('jenis', 'pengeluaran');

        if ($karangTarunaId !== null) {
            $pemasukanBuilder->where('karang_taruna_id', $karangTarunaId);
            $pengeluaranBuilder->where('karang_taruna_id', $karangTarunaId);
        }
        
        $pemasukan = $pemasukanBuilder->get()->getRow()->nominal ?? 0;
        $pengeluaran = $pengeluaranBuilder->get()->getRow()->nominal ?? 0;
        
        return $pemasukan - $pengeluaran;
    }
}

File: mobile/lib/models/kas_model.dart
class KasModel {
  final int id;
  final String jenis;
  final int nominal;
  final String keterangan;
  final String tanggal;
  final int dibuatOleh;
  final String? pembuat;
  final String createdAt;

  KasModel({
    required this.id,
    required this.jenis,
    required this.nominal,
    required this.keterangan,
    required this.tanggal,
    required this.dibuatOleh,
    this.pembuat,
    required this.createdAt,
  });

  bool get isPemasukan => jenis == 'pemasukan';

  factory KasModel.fromJson(Map<String, dynamic> json) {
    return KasModel(
      id: json['id'] != null ? (json['id'] is int ? json['id'] : int.tryParse(json['id'].toString()) ?? 0) : 0,
      jenis: json['jenis'] ?? '',
      nominal: json['nominal'] != null ? (json['nominal'] is int ? json['nominal'] : int.tryParse(json['nominal'].toString()) ?? 0) : 0,
      keterangan: json['keterangan'] ?? '',
      tanggal: json['tanggal'] ?? '',
      dibuatOleh: json['dibuat_oleh'] != null ? (json['dibuat_oleh'] is int ? json['dibuat_oleh'] : int.tryParse(json['dibuat_oleh'].toString()) ?? 0) : 0,
      pembuat: json['pembuat'],
      createdAt: json['created_at'] ?? '',
    );
  }
}
